### ISSUE TITLE
GroupUserCallForwardingSettingsGetListRequest and GroupUserCallForwardingSettingsGetListRequest - Server Error: [Error 6004] OCI XML Request validation error

### ISSUE THREAD

**Issue by fastcloudza** [cwmiller/broadworks-connector#65]
Hi,

These two requests are passing the validator in the library, but failing validation at the server.  Cant figure out why I keep getting a:
Server Error: [Error 6004] OCI XML Request validation error 

Requests for GroupUserCallForwardingSettingsGetListRequest and EnterpriseUserCallForwardingSettingsGetListRequest  are similar so going to post the former only. 

// set paging response
$ResponsePagingControl = (new ResponsePagingControl())
    ->setResponseStartIndex(1)
    ->setResponsePageSize(2000);

// sort by extension
$SortByExtension = (new SortByExtension())
    ->setIsAscending(true);

$request1 = (new GroupUserCallForwardingSettingsGetListRequest())
    ->setServiceProviderId($service_provider_id)
    ->setGroupId($xsp_user_group)
    ->setCallForwardingService(CallForwardingService::CALL_FORWARDING_ALWAYS())
    ->setResponsePagingControl($ResponsePagingControl)
    ->setSortByExtension($SortByExtension);

Result:
Server Error: [Error 6004] OCI XML Request validation error

Here's the var_dump of $request1;
object(CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest)#7 (20) { ["serviceProviderId":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> string(8) "XXXXXXX" ["groupId":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> string(12) "1-XXXXXX" ["callForwardingService":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> object(CWM\BroadWorksConnector\Ocip\Models\CallForwardingService)#8 (1) { ["value":protected]=> string(22) "Call Forwarding Always" } ["responsePagingControl":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> object(CWM\BroadWorksConnector\Ocip\Models\ResponsePagingControl)#5 (2) { ["responseStartIndex":"CWM\BroadWorksConnector\Ocip\Models\ResponsePagingControl":private]=> int(1) ["responsePageSize":"CWM\BroadWorksConnector\Ocip\Models\ResponsePagingControl":private]=> int(2000) } ["sortByUserLastName":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> NULL ["sortByUserFirstName":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> NULL ["sortByUserDepartment":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> NULL ["sortByDn":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> NULL ["sortByExtension":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> object(CWM\BroadWorksConnector\Ocip\Models\SortByExtension)#6 (2) { ["isAscending":"CWM\BroadWorksConnector\Ocip\Models\SortCriteria":private]=> bool(true) ["isCaseSensitive":"CWM\BroadWorksConnector\Ocip\Models\SortCriteria":private]=> NULL } ["sortByForwardedToNumber":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> NULL ["includeVirtualUsers":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> NULL ["searchCriteriaUserLastName":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> array(0) { } ["searchCriteriaUserFirstName":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> array(0) { } ["searchCriteriaUserId":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> array(0) { } ["searchCriteriaDn":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> array(0) { } ["searchCriteriaExtension":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> array(0) { } ["searchCriteriaEmailAddress":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> array(0) { } ["searchCriteriaExactUserDepartment":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> NULL ["searchCriteriaExactUserInTrunkGroup":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> NULL ["searchCriteriaForwardedToNumber":"CWM\BroadWorksConnector\Ocip\Models\GroupUserCallForwardingSettingsGetListRequest":private]=> array(0) { } }

What am I doing wrong?   Thank you.

**Comment by cwmiller** [comment]
Thank you for reporting this!

It's a bug within the library that's preventing the inherited properties on `SortByExtension` from being included in the serialized XML. I'm working on a fix.

**Comment by fastcloudza** [comment]
Thank you.

**Comment by fastcloudza** [comment]
As a follow-up, and notwithstanding your reference to a bug:
1) I tested all sort functions (sortByUserLastName, sortByUserFirstName, sortByUserDepartment, sortByDn, sortByExtension, sortByForwardedToNumber) and get the same "Server Error: [Error 6004] OCI XML Request validation error".  
2) Tested using the GroupUserCallingPlanSettingsGetListRequest class, and get the same error.  
Possible the bug affects all the sort functions?  Possible that I'm getting something wrong? 

I have attached additional information.
[GroupUserCallingPlanSettingsGetListRequest_setSortByExtension.txt](https://github.com/cwmiller/broadworks-connector/files/8796032/GroupUserCallingPlanSettingsGetListRequest_setSortByExtension.txt)
[GroupUserCallForwardingSettingsGetListRequest_setSortByUserFirstName.txt](https://github.com/cwmiller/broadworks-connector/files/8796033/GroupUserCallForwardingSettingsGetListRequest_setSortByUserFirstName.txt)

**Comment by fastcloudza** [comment]
As a follow-up, and notwithstanding your reference to a bug:
1) I tested all sort functions (sortByUserLastName, sortByUserFirstName, sortByUserDepartment, sortByDn, sortByExtension, sortByForwardedToNumber) and get the same "Server Error: [Error 6004] OCI XML Request validation error".  
2) Tested using the GroupUserCallingPlanSettingsGetListRequest class, and get the same error.  
Possible the bug affects all the sort functions?  Possible that I'm getting something wrong? 

I have attached additional information.

[GroupUserCallForwardingSettingsGetListRequest_setSortByUserFirstName.txt](https://github.com/cwmiller/broadworks-connector/files/8796047/GroupUserCallForwardingSettingsGetListRequest_setSortByUserFirstName.txt)
[GroupUserCallingPlanSettingsGetListRequest_setSortByExtension.txt](https://github.com/cwmiller/broadworks-connector/files/8796048/GroupUserCallingPlanSettingsGetListRequest_setSortByExtension.txt)
